packages/flutter_tools/lib/src/commands/generate_localizations.dart

```diff
@@ -207,57 +223,68 @@ class GenerateLocalizationsCommand extends FlutterCommand {
-    final List<String> preferredSupportedLocales = stringsArg('preferred-supported-locales');
-    final String? headerString = stringArgDeprecated('header');
-    final String? headerFile = stringArgDeprecated('header-file');
-    final bool useDeferredLoading = boolArgDeprecated('use-deferred-loading');
-    final String? inputsAndOutputsListPath = stringArgDeprecated('gen-inputs-and-outputs-list');
-    final bool useSyntheticPackage = boolArgDeprecated('synthetic-package');
-    final String? projectPathString = stringArgDeprecated('project-dir');
-    final bool areResourceAttributesRequired = boolArgDeprecated('required-resource-attributes');
-    final bool usesNullableGetter = boolArgDeprecated('nullable-getter');
+      precacheLanguageAndRegionTags();
 
-    precacheLanguageAndRegionTags();
+      try {
+        outputFileList = (LocalizationsGenerator(
+          fileSystem: _fileSystem,
+          inputPathString: inputPathString,
+          outputPathString: outputPathString,
+          templateArbFileName: templateArbFileName,
+          outputFileString: outputFileString,
+          classNameString: classNameString,
+          preferredSupportedLocales: preferredSupportedLocales,
+          headerString: headerString,
+          headerFile: headerFile,
+          useDeferredLoading: useDeferredLoading,
+          inputsAndOutputsListPath: inputsAndOutputsListPath,
+          useSyntheticPackage: useSyntheticPackage,
+          projectPathString: projectPathString,
+          areResourceAttributesRequired: areResourceAttributesRequired,
+          untranslatedMessagesFile: untranslatedMessagesFile,
+          usesNullableGetter: usesNullableGetter,
+          logger: _logger,
+        )
+          ..loadResources()
+          ..writeOutputFiles())
+          .outputFileList;
+      } on L10nException catch (e) {
+        throwToolExit(e.message);
+      }
+    }
 
-    try {
-      LocalizationsGenerator(
-        fileSystem: _fileSystem,
-        inputPathString: inputPathString,
-        outputPathString: outputPathString,
-        templateArbFileName: templateArbFileName,
-        outputFileString: outputFileString,
-        classNameString: classNameString,
-        preferredSupportedLocales: preferredSupportedLocales,
-        headerString: headerString,
-        headerFile: headerFile,
-        useDeferredLoading: useDeferredLoading,
-        inputsAndOutputsListPath: inputsAndOutputsListPath,
-        useSyntheticPackage: useSyntheticPackage,
-        projectPathString: projectPathString,
-        areResourceAttributesRequired: areResourceAttributesRequired,
-        untranslatedMessagesFile: untranslatedMessagesFile,
-        usesNullableGetter: usesNullableGetter,
-        logger: _logger,
-      )
-        ..loadResources()
-        ..writeOutputFiles();
-    } on L10nException catch (e) {
-      throwToolExit(e.message);
+    // All other post processing.
+    if (boolArg('format') ?? false) {
```

Comment:
This will only apply to the format when using flutter tools directly. Generalizing is also included in building targets.